core/domain/services/graphql/mutators/PriceDelete.php

```diff
@@ -33,8 +38,14 @@ public static function mutateAndGetPayload(EEM_Price $model, Price $type)
             try {
                 /** @var EE_Price $entity */
                 $entity = EntityMutator::getEntityFromInputData($model, $input);
+
                 // Delete the entity
-                $result = ! empty($input['deletePermanently']) ? $entity->delete_permanently() : $entity->delete();
+                if (! empty($input['deletePermanently'])) {
+                    $result = PriceDelete::deletePriceAndRelations($entity);
+                } else {
```

Comment:
👍 good thinking extracting this too... never know when we might need it somewhere else